aggregator/src/aggregator.rs

```diff
@@ -1883,6 +1883,33 @@ impl VdafOps {
             }
         };
 
+        // Check if any two extensions have the same extension type across public
+        // and private extension fields. If so, the Aggregator MUST mark the input
+        // share as invalid with error invalid_message. (§4.6.2.4 step 7)
+        //
+        // Note at this point we can't provide Error::InvalidMessage here because
+        // we're packaging this into a report rejection list.
+        let mut extensions = HashMap::new();
+        if !leader_private_extensions
+            .iter()
+            .chain(report.metadata().public_extensions())
+            .all(|extension| {
+                extensions
+                    .insert(*extension.extension_type(), extension.extension_data())
+                    .is_none()
+            })
+        {
+            debug!(
+                task_id = %task.id(),
+                report_id = ?report.metadata().id(),
+                "Received report share with duplicate extensions",
+            );
+            metrics
+                .upload_decode_failure_counter
+                .add(1, &[KeyValue::new("type", "duplicate_extension")]);
```

Comment:
We could add a `upload_decode_failure_counter.add(0, &[Keyvalue::new("type", "duplicate_extension")]);` in `Aggregator::new()` to eagerly populate this time series at startup.